components/salsa-macro-rules/src/setup_tracked_fn.rs

```diff
@@ -158,6 +158,15 @@ macro_rules! setup_tracked_fn {
 
                 const CYCLE_STRATEGY: $zalsa::CycleRecoveryStrategy = $zalsa::CycleRecoveryStrategy::$cycle_recovery_strategy;
 
+                unsafe fn maybe_update<'db>(old_pointer: *mut Self::Output<'db>, new_value: Self::Output<'db>) -> bool {
+                    unsafe {
+                        use $zalsa::UpdateFallback;
+                        salsa::plumbing::UpdateDispatch::<Self::Output<'db>>::maybe_update(
+                            old_pointer, new_value
+                        )
+                    }
+                }
```

Comment:
Given this is not called and just used as an comptime assertion, there isn't really a reason for us to make this a trait method is there? We can just emit it as a private inherent function here